gio/src/app_info.rs

```diff
@@ -93,10 +93,16 @@ impl<O: IsA<AppInfo>> AppInfoExtManual for O {
     #[cfg_attr(feature = "dox", doc(cfg(feature = "v2_60")))]
     fn launch_uris_async_future<P: IsA<AppLaunchContext> + Clone + 'static>(
         &self,
-        uris: &[&str],
+        uris: &[impl AsRef<str>],
         context: Option<&P>,
     ) -> Pin<Box_<dyn std::future::Future<Output = Result<(), glib::Error>> + 'static>> {
-        let uris = uris.iter().copied().map(String::from).collect::<Vec<_>>();
+        let uris = uris
+            .as_ref()
+            .iter()
+            .map(|v| v.as_ref())
+            .copied()
+            .map(String::from)
+            .collect::<Vec<_>>();
```

Comment:
`launch_uris_async` takes a `uris: &[impl AsRef<str>],` as well, it should be possible to pass this as-is or by simply copying the slice? Otherwise, this should perhaps take a `Cow` instead. Note that the closure below re-borrows the items.

Unfortunately I cannot compile-test and double-check this to give you better suggestions, the rest of the code fails to compile with a missing trait bound for `to_glib_none()` on `&[impl AsRef<str>]`.